shimmingtoolbox/shim/sequencer.py

```diff
@@ -42,838 +38,1063 @@
 }
 
 
-@timeit
-def shim_sequencer(nii_fieldmap, nii_anat, nii_mask_anat, slices, coils: ListCoil, method='least_squares',
-                   opt_criteria='mse', mask_dilation_kernel='sphere', mask_dilation_kernel_size=3, reg_factor=0,
-                   path_output=None):
-    """
-    Performs shimming according to slices using one of the supported optimizers and coil profiles.
-
-    Args:
-        nii_fieldmap (nibabel.Nifti1Image): Nibabel object containing fieldmap data in 3d and an affine transformation.
-        nii_anat (nibabel.Nifti1Image): Nibabel object containing anatomical data in 3d.
-        nii_mask_anat (nibabel.Nifti1Image): 3D anat mask used for the optimizer to shim in the region of interest.
+class Sequencer(object):
+    def _select_optimizer(self, unshimmed, affine, pmu: PmuResp = None):
```

Comment:
"\_" designate private methods. Since this is now a class, we should make it available in the object. This would allow to change the optimizer instead of recreating a new object. So remove the "_" and make sure that calling it indeed does correctly change the optimizer.